elements/rh-navigation-primary/test/rh-navigation-primary.spec.ts

```diff
@@ -69,32 +142,83 @@ describe('<rh-navigation-primary>', function() {
         expect(element.shadowRoot?.querySelector('#container')?.classList.contains('compact')).to.be.true;
       });
 
+      it('menu should be closed', async function() {
+        const snapshot = await a11ySnapshot();
+        expect(snapshot).axQuery({ name: 'Menu' }).to.not.have.property('expanded', true);
+      });
+
+      it('should hide content behind hamburger menu', async function() {
+        const snapshot = await a11ySnapshot();
+        expect(snapshot).to.not.have.axQuery({ name: HamburgerItem1Name });
+        expect(snapshot).to.not.have.axQuery({ name: HamburgerItem2Name });
+        expect(snapshot).to.not.have.axQuery({ name: HamburgerItem3Name });
+        expect(snapshot).to.not.have.axQuery({ name: HamburgerItem4Name });
+        expect(snapshot).to.not.have.axQuery({ name: HamburgerItem5Name });
+        expect(snapshot).to.not.have.axQuery({ name: 'Link 1' });
+        expect(snapshot).to.not.have.axQuery({ name: 'Link 2' });
+        expect(snapshot).to.not.have.axQuery({ name: 'Link 3' });
+      });
+
+      it('should not hide right-side content behind hamburger menu', async function() {
+        const snapshot = await a11ySnapshot();
+        expect(snapshot).to.have.axQuery({ name: 'Item 6' });
+        expect(snapshot).to.have.axQuery({ name: 'Item 7' });
+        expect(snapshot).to.have.axQuery({ name: 'Item 8' });
+      });
+
       describe('interactions', function() {
         describe('toggling hamburger menu', function() {
-          let details: HTMLDetailsElement | null | undefined;
           let summary: HTMLElement | null | undefined;
 
           beforeEach(async function() {
-            details = element.shadowRoot?.querySelector('#hamburger');
+            const details = element.shadowRoot?.querySelector('#hamburger');
             summary = details?.querySelector('summary');
             summary?.click();
```

Comment:
replace these with keyboard presses i.e tab sequences

- avoids brittle shadow queries
- as a bonus, we gain keyboard a11y testing

Reply:
https://github.com/RedHat-UX/red-hat-design-system/issues/2408